include/openenclave/internal/report.h

```diff
@@ -15,42 +15,30 @@
 /*
 **==============================================================================
 **
-** _oe_get_revocation_info_args
+** _oe_get_sgx_quote_verification_collateral_args
 **
 **==============================================================================
 */
-typedef struct _oe_get_revocation_info_args
+typedef struct _oe_get_sgx_quote_verification_collateral_args
 {
-    oe_result_t result;              /* out */
-    uint8_t fmspc[6];                /* in */
-    const char* crl_urls[3];         /* in */
-    uint32_t num_crl_urls;           /* in */
-    uint8_t* tcb_info;               /* out */
-    size_t tcb_info_size;            /* out */
-    uint8_t* tcb_issuer_chain;       /* out */
-    size_t tcb_issuer_chain_size;    /* out */
-    uint8_t* crl[3];                 /* out */
-    size_t crl_size[3];              /* out */
-    uint8_t* crl_issuer_chain[3];    /* out */
-    size_t crl_issuer_chain_size[3]; /* out */
-    uint8_t* buffer;                 /* out */
-} oe_get_revocation_info_args_t;
-
-/*
-**==============================================================================
-**
-** _oe_get_qe_identity_info_args
-**
-**==============================================================================
-*/
-typedef struct _oe_get_qe_identity_info_args
-{
-    uint8_t* qe_id_info;      /* out */
-    size_t qe_id_info_size;   /* out */
-    uint8_t* issuer_chain;    /* out */
-    size_t issuer_chain_size; /* out */
-    uint8_t* host_out_buffer; /* out */
-} oe_get_qe_identity_info_args_t;
+    oe_result_t result;                   /* out */
```

Comment:
When the caller does not pass in the endorsements/collaterals in oe_verify_report(), we currently fetch the serialized collaterals and then use the serialized collaterals for verification.  One optimization, given that there is so much data to copy, is to NOT create the serialized collaterals in this specific scenario and just use the collaterals straight from the DCAP client.  In the scenario when the caller passes in collaterals, we would need to convert the serializable object to something like this struct (reverse of create_sgx_endorsements()).

Reply:
Thanks for the suggestion, will open an issue and address them later.